test/e2e/scheduling/priorities.go

```diff
@@ -0,0 +1,713 @@
+		Expect(success).To(Equal(true))
+
+		defer framework.RemoveAvoidPodsOffNode(cs, nodeName)
+
+		By(fmt.Sprintf("Scale the RC: %s to len(nodeList.Iten)-1 : %v.", rc.Name, len(nodeList.Items)-1))
+
+		framework.ScaleRC(f.ClientSet, f.InternalClientset, ns, rc.Name, uint(len(nodeList.Items)-1), true)
+		testPods, err := cs.Core().Pods(ns).List(metav1.ListOptions{
+			LabelSelector: "name=scheduler-priority-avoid-pod",
+		})
+		Expect(err).NotTo(HaveOccurred())
+		By(fmt.Sprintf("Verify the pods should not scheduled to the node: %s", nodeName))
+		for _, pod := range testPods.Items {
+			Expect(pod.Spec.NodeName).NotTo(Equal(nodeName))
+		}
+	})
+
+	It("Pod should perfer to scheduled to nodes pod can tolerate", func() {
+		// make the nodes have balanced cpu,mem usage ratio
+		createBalancedPodForNodes(f, cs, ns, nodeList.Items, podRequestedResource, 0.5)
+		//we need apply more taints on a node, because one match toleration only count 1
+		By("Trying to apply 10 taint on the nodes except first one.")
+		nodeName := nodeList.Items[0].Name
+
+		for index, node := range nodeList.Items {
+			if index == 0 {
+				continue
+			}
+			for i := 0; i < 10; i++ {
+				testTaint := addRandomTaitToNode(cs, node.Name)
+				defer framework.RemoveTaintOffNode(cs, node.Name, *testTaint)
+			}
+		}
+		By("Create a pod without any tolerations")
+		tolerationPodName := "without-tolerations"
+		pod := createPausePod(f, pausePodConfig{
+			Name: tolerationPodName,
+		})
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+
+		By("Pod should prefer scheduled to the node don't have the taint.")
+		tolePod, err := cs.Core().Pods(ns).Get(tolerationPodName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(tolePod.Spec.NodeName).To(Equal(nodeName))
+
+		By("Trying to apply 10 taint on the first node.")
+		var tolerations []v1.Toleration
+		for i := 0; i < 10; i++ {
+			testTaint := addRandomTaitToNode(cs, nodeName)
+			tolerations = append(tolerations, v1.Toleration{Key: testTaint.Key, Value: testTaint.Value, Effect: testTaint.Effect})
+			defer framework.RemoveTaintOffNode(cs, nodeName, *testTaint)
+		}
+		tolerationPodName = "with-tolerations"
+		By("Create a pod that tolerates all the taints of the first node.")
+		pod = createPausePod(f, pausePodConfig{
+			Name:        tolerationPodName,
+			Tolerations: tolerations,
+		})
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+
+		By("Pod should prefer scheduled to the node that pod can tolerate.")
+		tolePod, err = cs.Core().Pods(ns).Get(tolerationPodName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(tolePod.Spec.NodeName).To(Equal(nodeName))
+	})
+})
+
+// createBalancedPodForNodes creates a pod per node that asks for enough resources to make all nodes have the same mem/cpu usage ratio.
+func createBalancedPodForNodes(f *framework.Framework, cs clientset.Interface, ns string, nodes []v1.Node, requestedResource *v1.ResourceRequirements, ratio float64) {
```

Comment:
Similar does not mean same, as before we run the test, there really have a lot system pods make the resource not balanced.